update Share doc with functional component example

amandeepmittal opened the pull request:
Here is an update for the #1579 regarding the Share component doc.

The class component already exists. I am adding a functional component example too as well as add the support for Snack player.

<!--
Thank you for the PR! Contributors like you keep React Native awesome!

Please see the Contribution Guide for guidelines:

https://github.com/facebook/react-native-website/blob/master/CONTRIBUTING.md

If your PR references an existing issue, please add the issue number below:

#<Issue>
-->

Simek commented on docs/share.md:
````diff
@@ -3,6 +3,85 @@ id: share
+<block class="functional syntax" />
+
+```SnackPlayer name=Function%20Component%20Example
+import React, { Component } from "react";
````
Unnecessary `Component` import.

Simek commented on docs/share.md:
````diff
@@ -3,6 +3,85 @@ id: share
+```SnackPlayer name=Function%20Component%20Example
+import React, { Component } from "react";
+import { Share, Button } from "react-native";
+const ShareExample = () => {
````
I think that `export default ShareExample ...` will be better for the consistency between examples.

amandeepmittal commented:
Hey @Simek thanks for the review. I made an update to the doc and also (as per Code Tips) am adding `&supportedPlatforms=ios,android` to each Snack Player to disable the support for `web`. Is this okay, let me know, otherwise I will make changes.

amandeepmittal commented:
Hey @Simek thanks for the review. I made an update to the doc and also (as per Code Tips) am adding `&supportedPlatforms=ios,android` to each Snack Player to disable the support for `web`. Is this okay, let me know, otherwise I will make changes.

Simek commented:
@amandeepmittal It's the right way, thank you!

I have also left one more comment which should be addressed before merge.

Simek commented on docs/share.md:
```diff
@@ -3,6 +3,86 @@ id: share
+import React from "react";
+import { Share, Button } from "react-native";
+
+export default const ShareExample = () => {
```
This syntax is invalid and produce error. I have deliberate posted the right way. Please test you examples before committing.

<img width="193" alt="Annotation 2020-02-27 162156" src="https://user-images.githubusercontent.com/719641/75457891-58927a00-597d-11ea-882c-6d8e0037c381.png">

amandeepmittal commented:
Sorry about that @Simek.

I have changed the example a bit not the core functionality. I have added a `View` with spacing such that someone running the example using Expo Snack on a device with notch will get the desired results. Let me know if this okay?

Also, I have tested both the examples in the Snack player.

Simek commented on docs/share.md:
```diff
@@ -3,6 +3,97 @@ id: share
+  };
+  return (
+    <View style={{ marginTop: 50 }}>
+      <Button onPress={() => onShare()} title="Share" />
```
This part can be simplified to `onPress={onShare}`.

amandeepmittal replied:
Done.

Simek commented:
@amandeepmittal Other than comment above LGTM, but I would prefer to have `export default` at the top of the code as I mentioned earlier. This matches style of most of new examples, reduce possibility of issues while copying and saves few lines.

amandeepmittal commented:
@Simek thank you.

If I use the `export default` at the top in the functional component example, I am getting the following error in the Snack. So either I have to use the keyword `function` or have to eliminate the name of the function and it could be like `export default () => {...}`.

Let me know what would be the optimal preference?

![Screenshot 2020-02-28 at 5 31 00 PM](https://user-images.githubusercontent.com/10234615/75547106-34539d80-5a50-11ea-9112-6be2554a163f.png)

Simek commented:
@amandeepmittal It's just: `export default ShareExample = () => {`, without `const`, as my first review comment stated.

amandeepmittal commented:
Thanks @Simek I am sure have taken a lot of your time for this one.

Simek commented:
@amandeepmittal I hope this process will help you with the future PRs. 

And on more thing, `export default` position in class component needs to be reverted back to the top, as in original example, because now we are again not consistent like at beginning of the review.
